apps/backend-e2e/src/map-review.e2e-spec.ts

```diff
@@ -0,0 +1,249 @@
+        await req.expandTest({
+          url: `map-review/${review.id}`,
+          token: token,
+          expand: 'reviewer',
+          validate: MapReviewDto
+        });
+      });
+
+      it('should return the requested review including map information', async () => {
+        await req.expandTest({
+          url: `map-review/${review.id}`,
+          token: token,
+          expand: 'map',
+          validate: MapReviewDto
+        });
+      });
+
+      // Test that permissions checks are getting called
+      it('should 403 if the user does not have permission to access to the map', async () => {
+        await prisma.mMap.update({
+          where: { id: map.id },
+          data: { status: MapStatusNew.PRIVATE_TESTING }
+        });
+
+        await req.get({
+          url: `map-review/${review.id}`,
+          status: 403,
+          token: token
+        });
+
+        await prisma.mMap.update({
+          where: { id: map.id },
+          data: { status: MapStatusNew.APPROVED }
+        });
+      });
+
+      it('should return 404 for a nonexistent review', () =>
+        req.get({
+          url: `map-review/${NULL_ID}`,
+          status: 404,
+          token: token
+        }));
+
+      it('should 401 when no access token is provided', () =>
+        req.unauthorizedTest('map-review/1', 'get'));
+    });
+
+    describe('DELETE', () => {
+      let user, token, u2Token, modToken, map, review;
+      const assetPath = mapReviewAssetPath('1');
+
+      beforeAll(async () => {
+        [[user, token], modToken] = await Promise.all([
+          db.createAndLoginUser(),
+          db.createAndLoginUser(),
+          db.loginNewUser({ data: { roles: Role.MODERATOR } })
+        ]);
+
+        map = await db.createMap({
+          status: MapStatusNew.PUBLIC_TESTING
+        });
+      });
+
+      afterAll(() => db.cleanup('mMap', 'user'));
+
+      beforeEach(async () => {
+        review = await prisma.mapReview.create({
+          data: {
+            mainText: 'This map was E Z',
+            imageIDs: ['1'],
+            suggestions: [
+              {
+                track: 1,
+                gamemode: Gamemode.SURF,
+                tier: 1,
+                comment: 'I surfed this backwards',
+                gameplayRating: 10
+              }
+            ],
+            mmap: { connect: { id: map.id } },
+            reviewer: { connect: { id: user.id } },
+            resolved: true
+          }
+        });
+
+        await fileStore.add(assetPath, Buffer.alloc(1024));
+      });
+
+      afterEach(async () => {
+        await fileStore.delete(assetPath);
+        await db.cleanup('mapReview');
+      });
+
+      it('should allow a user to delete their own review', () =>
+        req.del({
+          url: `map-review/${review.id}`,
+          status: 204,
+          token: token
+        }));
+
+      it('should not allow another user to delete a own review', () =>
```

Comment:
"should not allow another user to delete someone else's review" maybe?

Reply:
Fixed